iib/workers/tasks/legacy.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# This file can be deleted once OMPS is retired
+import logging
+import os
+import shutil
+import tempfile
+
+import requests
+from operatorcourier import api, errors
+
+from iib.exceptions import IIBError
+from iib.workers.api_utils import get_request, set_request_state, update_request
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import run_cmd, skopeo_inspect
+
+log = logging.getLogger(__name__)
+
+
+def get_legacy_support_packages(bundles):
+    """
+    Checks if legacy support is required for any packages in the bundles.
+
+    :param list<str> bundles: a list of strings representing the pull specifications of the bundles
+        to add to the index image being built.
+    :return: a set of packages that require legacy support
+    :rtype: set
+    """
+    packages = set()
+    for bundle in bundles:
+        skopeo_out = skopeo_inspect(f'docker://{bundle}')
+        if skopeo_out['Labels'].get('com.redhat.delivery.backport', False):
+            packages.add(skopeo_out['Labels']['operators.operatorframework.io.bundle.package.v1'])
+
+    return packages
+
+
+def _podman_login():
+    """
+    Login into the registry to be able to access private repos.
+
+    :raises iib.exceptions.IIBError: if the login fails
+    """
+    conf = get_worker_config()
+    log.info('Logging in %s using podman', conf['iib_registry'])
+    username, password = conf['iib_registry_credentials'].split(':')
```

Comment:
This is kind of odd. Can we just mount the secret file on the host? Check out `~/.docker/config.json` in your host.

Reply:
As discussed, we will fix this in another PR next week.